webull/webull.py

```diff
@@ -1046,6 +1045,39 @@ def get_bars(self, stock=None, tId = None, interval='m1', count=1, extendTrading
             df.loc[to_datetime(datetime.fromtimestamp(int(row[0])).astimezone(time_zone))] = data
         return df.iloc[::-1]
 
+    def get_options_bars(self, derivativeId=None, interval='1m', count=1, timeStamp=None):
+        '''
+        get bars returns a pandas dataframe
+        params:
+            interval:1m, 5m, 30m, 60m, 1d
+            count: number of bars to return
+            timeStamp: If epoc timestamp is provided, return bar count up to timestamp. If not set default to current time.
```

Comment:
you've neglected to supply the "direction" parameter. I found that if that parameter is missing or set to anything except "1", the time stamp is ignored and all data for for the derivate (up to count) is returned

Reply:
Yep, you're right! I've made the changes, thanks for pointing that out!
I also found that if the direction is set to 1, it ignores the `count` parameter and returns all bars on and after the timestamp